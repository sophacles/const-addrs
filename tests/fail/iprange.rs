use const_addrs::iprange;

fn main() {
    let _ = iprange!("2001:db8::32:23", "2001:db8::32::ffff");
    let _ = iprange!("2001:db8::32::23", "2001:db8::32:ffff");
    let _ = iprange!("2001:db8::32::23", "2001:db8::32::ffff");
    let _ = iprange!("2001:db8::32:23", "10.0.0.2");
    let _ = iprange!("10.0.0.1", "2001:db8::32:23");
    let _ = iprange!("2001:db8::32:23", "10.0.0.1");
    let _ = iprange!("100.0.0", "10.0.0.255");
    let _ = iprange!("10.0.0.0", "10.0.0255");
}
