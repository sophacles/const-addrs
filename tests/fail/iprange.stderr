error: invalid IP address syntax
 --> tests/fail/iprange.rs:4:41
  |
4 |     let _ = iprange!("2001:db8::32:23", "2001:db8::32::ffff");
  |                                         ^^^^^^^^^^^^^^^^^^^^

error: invalid IP address syntax
 --> tests/fail/iprange.rs:5:22
  |
5 |     let _ = iprange!("2001:db8::32::23", "2001:db8::32:ffff");
  |                      ^^^^^^^^^^^^^^^^^^

error: macro expansion ignores `compile_error` and any tokens following
 --> tests/fail/iprange.rs:6:42
  |
6 |     let _ = iprange!("2001:db8::32::23", "2001:db8::32::ffff");
  |             -----------------------------^^^^^^^^^^^^^^^^^^^^- caused by the macro expansion here
  |
  = note: the usage of `iprange!` is likely invalid in expression context

error: invalid IP address syntax
 --> tests/fail/iprange.rs:6:22
  |
6 |     let _ = iprange!("2001:db8::32::23", "2001:db8::32::ffff");
  |                      ^^^^^^^^^^^^^^^^^^

error: Addr is IPv4 but IPv6 is expected
 --> tests/fail/iprange.rs:7:41
  |
7 |     let _ = iprange!("2001:db8::32:23", "10.0.0.2");
  |                                         ^^^^^^^^^^

error: Addr is IPv6 but IPv4 is expected
 --> tests/fail/iprange.rs:8:34
  |
8 |     let _ = iprange!("10.0.0.1", "2001:db8::32:23");
  |                                  ^^^^^^^^^^^^^^^^^

error: Addr is IPv4 but IPv6 is expected
 --> tests/fail/iprange.rs:9:41
  |
9 |     let _ = iprange!("2001:db8::32:23", "10.0.0.1");
  |                                         ^^^^^^^^^^

error: invalid IP address syntax
  --> tests/fail/iprange.rs:10:22
   |
10 |     let _ = iprange!("100.0.0", "10.0.0.255");
   |                      ^^^^^^^^^

error: invalid IP address syntax
  --> tests/fail/iprange.rs:11:34
   |
11 |     let _ = iprange!("10.0.0.0", "10.0.0255");
   |                                  ^^^^^^^^^^^
